src/domain/users/users.repository.spec.ts

```diff
@@ -63,4 +72,231 @@ describe('UsersRepository', () => {
       expect(result).toEqual({ id: mockUser.id });
     });
   });
+
+  describe('deleteWalletFromUser', () => {
+    it('should remove a wallet from a user', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const addressToRemove = getAddress(faker.finance.ethereumAddress());
+
+      const mockWallets = [walletBuilder().build(), walletBuilder().build()];
+      const mockUser = userBuilder().with('wallets', mockWallets).build();
+      mockUserRepository.findOne.mockResolvedValueOnce(mockUser);
+      mockWalletRepository.delete.mockResolvedValue({ affected: 1, raw: {} });
+
+      await usersRepository.deleteWalletFromUser({
+        authPayload,
+        walletAddress: addressToRemove,
+      });
+
+      expect(mockWalletRepository.delete).toHaveBeenCalledWith({
+        user: { id: mockUser.id },
+        address: addressToRemove,
+      });
+    });
+
+    it('should throw an UnauthorizedException if the auth payload is empty', async () => {
+      const authPayload = new AuthPayload();
```

Comment:
Nit: I think it'd be clearer if we use the `authPayloadBuilder` and explicitly define `signer_address` as `undefined`.